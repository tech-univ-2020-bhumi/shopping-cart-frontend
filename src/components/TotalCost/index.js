import React from 'react';
import propTypes from 'prop-types';
import * as styles from './index.module.css';
import Button from '../Button';

function TotalCost(props) {
  const { total, onClickHandler } = props;

  return (
    <div className={styles.totalCard}>
      <div className={styles.totalCardText}>
        <div className={styles.total}>
          Total
        </div>
        <div className={styles.cost}>
          RS
        </div>
        <div className={styles.price}>
          {total}
        </div>
      </div>
      <hr />
      <div className={styles.totalCheckout}>
        <Button buttonColor="yellow" text="Checkout" onClickHandler={onClickHandler} />
      </div>
    </div>

  );
}
TotalCost.propTypes = {
  total: propTypes.number.isRequired,
  onClickHandler: propTypes.func.isRequired,
};

export default TotalCost;
