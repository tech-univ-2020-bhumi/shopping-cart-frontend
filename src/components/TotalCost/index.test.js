import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TotalCost from './index';


describe('the TotalCost component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<TotalCost />);

    expect(asFragment()).toMatchSnapshot();
  });
  it('should immediately call onclick handler when checkout is clicked', () => {
    const onClickMock = jest.fn();
    const { getByTestId } = render(<TotalCost onClickHandler={onClickMock} />);

    fireEvent.click(getByTestId('test-btn'));
    expect(onClickMock).toHaveBeenCalled();
  });
});
