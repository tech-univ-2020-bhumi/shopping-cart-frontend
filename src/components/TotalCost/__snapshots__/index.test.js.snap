// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the TotalCost component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        Total
      </div>
      <div>
        RS
      </div>
      <div />
    </div>
    <hr />
    <div>
      <button
        data-testid="test-btn"
        type="button"
      >
        Checkout
      </button>
    </div>
  </div>
</DocumentFragment>
`;
